backend/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -9,576 +10,553 @@
-        elif repos and channel_label in repos:
-            repo = repos[channel_label]
-            # In case we are using Repo object based on channel config, override it's id to name of the repo
-            # To not create channel directories in cache directory
-            repo.id = name
-        else:
-            # Not using values from config files
-            repo = yum.yumRepo.YumRepository(name)
-            repo.populate(self.configparser, name, self.yumbase.conf)
-        self.repo = repo
-
-        self.setup_repo(repo, no_mirrors, ca_cert_file, client_cert_file, client_key_file)
+                if zypper_cfg.has_option(section_name, option='proxy'):
+                    self.proxy_hostname = zypper_cfg.get(section_name, option='proxy')
+                    self.proxy_url = "http://%s" % self.proxy_hostname
+
+                if zypper_cfg.has_option(section_name, 'proxy_username'):
+                    self.proxy_user = zypper_cfg.get(section_name, 'proxy_username')
+
+                if zypper_cfg.has_option(section_name, 'proxy_password'):
+                    self.proxy_pass = zypper_cfg.get(section_name, 'proxy_password')
+
+        # Make sure baseurl ends with / and urljoin will work correctly
+        self.urls = [url]
+        if self.urls[0][-1] != '/':
+            self.urls[0] += '/'
+
+        # Exclude non-valid characters from reponame
+        self.reponame = self.name
+        for chr in ["$", " ", ".", ";"]:
+            self.reponame = self.reponame.replace(chr, "_")
+        self.channel_label = channel_label
+        # SUSE vendor repositories belongs to org = NULL
+        root = os.path.join(CACHE_DIR, str(org or "NULL"), self.channel_label or self.reponame)
+
+        self.repo = ZypperRepo(root=root, url=self.url, org=self.org)
         self.num_packages = 0
         self.num_excluded = 0
         self.gpgkey_autotrust = None
-        #    if '?' not in url:
-        #        real_urls.append(url)
-        #self.repo.urls = real_urls
         self.groupsfile = None
 
-    def __del__(self):
-        # close log files for yum plugin
-        for handler in logging.getLogger("yum.filelogging").handlers:
-            handler.close()
-        self.repo.close()
-
-    def _authenticate(self, url):
-        pass
+    def setup_repo(self, repo):
+        """
+        Setup repository and fetch metadata
+        """
+        self.salt = ZyppoSync(root=repo.root)
+        zypp_repo_url = self._prep_zypp_repo_url(self.url)
+
+        #REMOVE: Manually call Zypper
+        repo_cfg = '''[{reponame}]
+enabled=1
+autorefresh=0
+baseurl={baseurl}
+type=rpm-md
+'''
+        with open(os.path.join(repo.root, "etc/zypp/repos.d", str(self.channel_label or self.reponame) + ".repo"), "w") as repo_conf_file:
+            repo_conf_file.write(repo_cfg.format(reponame=self.channel_label or self.reponame, baseurl=zypp_repo_url))
+        os.system("zypper --root {} --gpg-auto-import-keys ref".format(repo.root))
```

Comment:
"autoimport" is a security problem. 

Correct would be:

- configure the correct keyring to use for checking (/var/lib/spacewalk/gpgdir)
- if there is an error because of a not trusted key, we need to ask the user to confirm.
- if the user run this command in a terminal we can ask to trust and if he do so, we import the key into the keyring.

Never "autotrust" gpg keys.

Reply:
@mcalmer this is fixed though, but these pieces will be anyway moved to Salt in the next iteration.

Reply:
@mcalmer - I've updated the PR to fix to address your request about the GPG keys:

- Do not auto-accept GPG keys.
- We ask the user to confirm for not trusted GPG keys.
- Allow "non-interactive" mode. No GPG are accepted.
- The old `/var/lib/spacewalk/gpgdir` (yum based keyring) is not longer used. New keyring is managed by Zypper in its own custom RPM database for reposync.

There are still some failures in the unit tests because of the new mocking. I'll try to fix those failures by today but, apart from that, this can be reviewed again. Thanks!